### ISSUE TITLE
TableNode has weird behaviour when using with iOS 11 UIContextualAction

### ISSUE THREAD

**Issue by huy-le** [TextureGroup/Texture#644]
// Texture version: 2.5

### TL;DR 
UIContextualAction won't work well with Texture's TableNode

### Key question
How can I tell TableNode that I did update data without performBatchUpdate / reloadData?

========
There are 2 issues:

### 1
Firstly, iOS 11, the delegate below is deprecated then replaced by 2 new delegates.
```swift
func tableView(_ tableView: UITableView, editActionsForRowAt indexPath: IndexPath) -> [UITableViewRowAction]?
``` 

```swift
func tableView(_ tableView: UITableView, leadingSwipeActionsConfigurationForRowAt indexPath: IndexPath) -> UISwipeActionsConfiguration?

func tableView(_ tableView: UITableView, trailingSwipeActionsConfigurationForRowAt indexPath: IndexPath) -> UISwipeActionsConfiguration
```
If you implemented 2 methods above, it will override the deprecated method, so in iOS 11, it never jumps to the deprecated one. However, TableNode does not have that feature, it always jumps to the deprecated.

To workarounds, I have to add UITableViewDelegate in the class, for example:

```swift
class TableNodeDelegate: NSObject, UITableViewDelegate, ASTableDataSource, ASTableDelegate {}
```

### 2
Secondly, because of unrespecting UIContextualAction, TableNode cannot use the new swipe action in iOS 11 for **destructive row**.

When we swipe to delete a row, UIContextualAction will handle the removing animation, therefore we only update dataSource, and never call performBatchUpdate to update the view (I tried it and it shows weird animation).

This is iOS 11 beta 2 release notes:
> When implementing a destructive UIContextualAction, delete the row in the data source and call the completion handler with true. **Do not call deleteRowsAtIndexPaths: in this case**.

Because Texture's DataController does not know what happens occurs in UIContextualAction, it not update its data. So if we scroll it around, the cell will display again.

**Comment by marcinrabieko** [comment]
Bump!

I have similar problem, my UITableViewRowAction appear double or triple times after swipe. It's really hard to create edit actions on swipe without trailingSwipeActionsConfigurationForRowAtIndexPath method. Any workaround pls?

<img width="186" alt="zrzut ekranu 2018-09-03 o 12 28 11" src="https://user-images.githubusercontent.com/6849618/44982066-db976780-af74-11e8-87e0-627c16901a86.png">

**Comment by Junglak** [comment]
Still nothing new?

**Comment by amarunko** [comment]
2019 guys, nothing new?

**Comment by phucdth12a** [comment]
Still nothing new in 2020 :(
